storage/mysql/mysql.go

```diff
@@ -113,24 +150,84 @@ func (s *Storage) ReadCheckpoint(ctx context.Context) ([]byte, error) {
 	}
 
 	var checkpoint []byte
-	if err := row.Scan(&checkpoint); err != nil {
+	var at int64
+	if err := row.Scan(&checkpoint, &at); err != nil {
 		if err == sql.ErrNoRows {
 			return nil, nil
 		}
-		return nil, err
+		return nil, fmt.Errorf("scan checkpoint: %v", err)
 	}
 	return checkpoint, nil
 }
 
-// writeCheckpoint stores the log signed checkpoint.
-func (s *Storage) writeCheckpoint(ctx context.Context, tx *sql.Tx, size uint64, rootHash []byte) error {
-	rawCheckpoint, err := s.newCheckpoint(size, rootHash)
+// publishCheckpoint creates a new checkpoint for the given size and root hash, and stores it in the
+// Checkpoint table.
+func (s *Storage) publishCheckpoint(ctx context.Context, i time.Duration) error {
```

Comment:
Rename `i` as `interval`? I took me a second to realise this is the checkpoint publish interval on L181.